src/metabase/lib/join.cljc

```diff
@@ -936,3 +938,34 @@
     (join-lhs-display-name-from-condition-lhs query stage-number join-or-joinable condition-lhs-column-or-nil)
     (join-lhs-display-name-for-first-join-in-first-stage query stage-number join-or-joinable)
     (i18n/tru "Previous results"))))
+
+(mu/defn join-condition-update-temporal-bucketing :- ::lib.schema.expression/boolean
+  "Updates the provided join-condition's fields' temporal-bucketing option, returns the updated join-condition.
+   Must be called on a standard join condition as per [[standard-join-condition?]].
+   This will sync both the lhs and rhs fields, and the fields that support the provided option will be updated.
+   Fields that do not support the provided option will be ignored."
+  ([query :- ::lib.schema/query
+    join-condition :- [:or ::lib.schema.expression/boolean ::lib.schema.common/external-op]
+    option-or-unit :- [:maybe [:or
+                               ::lib.schema.temporal-bucketing/option
+                               ::lib.schema.temporal-bucketing/unit]]]
+   (join-condition-update-temporal-bucketing query -1 join-condition option-or-unit))
+  ([query :- ::lib.schema/query
+    stage-number :- :int
+    join-condition :- [:or ::lib.schema.expression/boolean ::lib.schema.common/external-op]
+    option-or-unit :- [:maybe [:or
+                               ::lib.schema.temporal-bucketing/option
+                               ::lib.schema.temporal-bucketing/unit]]]
+   (let [[_ _ lhs rhs :as join-condition] (lib.common/->op-arg join-condition)]
+     (assert (standard-join-condition? join-condition)
+             (i18n/tru "Non-standard join condition. {0}" (pr-str join-condition)))
```

Comment:
I wonder if "standard join condition" is a concept users (should) understand. Translating the message makes sense if it is, and then it should be documented.